components/sup/src/sup_futures.rs

```diff
@@ -0,0 +1,48 @@
+//! Custom futures used throughout the Supervisor.
+
+use futures::{future::Either,
+              sync::oneshot,
+              Future,
+              IntoFuture};
+
+/// A handle attached to a future generated using
+/// `cancelable_future`. That future will be dropped when this handle
+/// is dropped.
+#[derive(Debug)]
+pub struct FutureHandle(oneshot::Sender<()>);
+
+impl FutureHandle {
+    /// A more explicit way to cancel a future instead of relying on
+    /// dropping the handle.
+    pub fn terminate(self) { let _ = self.0.send(()); }
```

Comment:
I find this to be a non-obvious way of ignoring the result, the shortest, simplest idiom seems to be
```suggestion
    pub fn terminate(self) { self.0.send(()).ok(); }
```
or just be explicit and throw in `#[allow(unused_must_use)]` since the function has only one call.